sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/src/main/java/com/azure/messaging/eventgrid/cloudnative/cloudevents/EventGridCloudNativeEventPublisher.java

```diff
@@ -0,0 +1,176 @@
+        if (event == null) {
+            return null;
+        }
+
+        // Identify data format by data content type
+        CloudEventDataFormat dataFormat;
+        final String dataContentType = event.getDataContentType();
+        if (dataContentType == null) {
+            dataFormat = CloudEventDataFormat.JSON;
+        } else {
+            // Future version:
+            // https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md#311-payload-serialization
+            // That is, a datacontenttype declares JSON-formatted content if its media type, when stripped of parameters,
+            // has the form */json or */*+json. If the datacontenttype is unspecified, processing SHOULD proceed as if
+            // the datacontenttype had been specified explicitly as application/json.
+
+            // Current version: https://github.com/cloudevents/sdk-java/blob/ff07dd83150deeb5eca5405cb80760b71e7b470c/
+            // formats/json-jackson/src/main/java/io/cloudevents/jackson/JsonFormat.java#L146-L149
+            final Matcher matcher = PATTERN.matcher(dataContentType);
+            if (matcher.find()) {
+                dataFormat = CloudEventDataFormat.JSON;
+            } else {
+                dataFormat = CloudEventDataFormat.BYTES;
+            }
+        }
+        final CloudEventData data = event.getData();
+        final BinaryData binaryData; // Create this variable to avoid NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
+        if (data != null) {
+            binaryData = BinaryData.fromBytes(data.toBytes());
+        } else {
+            binaryData = null;
+        }
+
+        // io.cloudevents.CloudEvent's id, source, type, and specversion are required.
+        // azure CloudEvent's source, type, and format(if data exist) are required.
+        final com.azure.core.models.CloudEvent cloudEvent = new com.azure.core.models.CloudEvent(
+            event.getSource().toString(), // required
+            event.getType(), // required
+            binaryData,
+            dataFormat,
+            event.getDataContentType() == null ? null : event.getDataContentType()
+        );
+
+        // optional: subject
+        if (event.getSubject() != null) {
+            cloudEvent.setSubject(event.getSubject());
+        }
+        // optional: data schema
+        final URI dataSchema = event.getDataSchema();
+        if (dataSchema != null) {
+            cloudEvent.setDataSchema(dataSchema.toString());
+        }
+        // optional: time
+        if (event.getTime() != null) {
+            cloudEvent.setTime(event.getTime());
+        }
+
+        cloudEvent.setId(event.getId()); // required
+        // SpecVersion is internally set by Azure CloudEvent to 1.0
+
+        // optional: extensionNames
+        event.getExtensionNames()
+            .stream()
+            .forEach(name -> cloudEvent.addExtensionAttribute(name, event.getAttribute(name)));
```

Comment:
We should check if `event.getAttribute(name)` returns a non-null value. What is the guidance for extension names that don't have attribute values?